### ISSUE TITLE
Enable order by while paging is enabled

### ISSUE THREAD

**Issue by Shaharsha** [logstash-plugins/logstash-input-jdbc#234]
Hi,
I'm using logstash-input-jdbc for pipelining data from Impala (JDBC).
My impala query returns alot of data (Hundreds of millions to billions), therefore i use pagination (jdbc_paging_enabled).
logstash-input-jdbc pagination wrappes the query with select * from (MY_QUERY) LIMIT X OFFSET Y.
My problem is that this syntax (select * from (MY_QUERY) LIMIT X OFFSET Y) is not supported in Impala.
Impala requires ORDER BY before the limit offset. For example: select * from (MY_QUERY) ORDER BY Z LIMIT X OFFSET Y.

Is there some existing solution for this problem?
My suggestion is adding an options called jdbc_paging_order_by_column, when it is set - it will add an order by before the LIMIT OFFSET.

Thanks,
Shahar

**Comment by mtdaniels** [comment]
I agree that this is a problem.  It also looks like the current implementation forces your query to be a subquery, then wraps it with the top/offset.  It would probably be more flexible to allow consumption of the paging and offset values as parameters.  This would allow the user to format the entire query as they want (including orderby, etc).

**Comment by dmitrymurashenkov** [comment]
One possible workaround is to use manual increments via sql_last_value.

For example:
```
select *
from my_object
where id > :sql_last_value
order by id asc
limit 1000
```
This way first you'll process data in batches of 1000 recording last value. 

Since if query has no ORDER BY clause DB can usually return data in arbitrary order there is no guarantees that paging will work correctly without ORDER BY in query anyway - it can process some data twice and not process some other rows.

**Comment by wlmqdz** [comment]
Any update? I think ORDER BY in the paging will be very helpful , especially when input large table and it needs to make sure all the data been processed.
